schematics/types/base.py

```diff
@@ -965,8 +966,39 @@ def to_primitive(self, value, context=None):
         else:
             value = value.astimezone(self.UTC)
         delta = value - self.EPOCH
-        ts = (delta.days * 24 * 3600) + delta.seconds + delta.microseconds / 1E6
-        return ts
+        return delta.total_seconds()
+
+
+class TimedeltaType(BaseType):
+
+    """Converts Python Timedelta objects into the corresponding value in seconds.
+    """
+
+    primitive_type = float
+    native_type = datetime.timedelta
+
+    MESSAGES = {
+        'convert': _("Couldn't interpret '{0}' value as Timedelta."),
+    }
+
+    def __init__(self, formats=None, **kwargs):
+        # type: (...) -> datetime.timedelta
+        super(TimedeltaType, self).__init__(**kwargs)
+
+    def _mock(self, context=None):
+        return datetime.timedelta(seconds=random.random() * 1000)
+
+    def to_native(self, value, context=None):
+        if isinstance(value, datetime.timedelta):
+            return value
+
+        try:
+            return datetime.timedelta(seconds=float(value))
+        except (ValueError, TypeError):
+            raise ConversionError(self.messages['convert'].format(value))
+
+    def to_primitive(self, value, context=None):
+        return value.total_seconds()
```

Comment:
Must return the value according to the selected init option here. Can divide by `timedelta(**{self.precision: 1})` to get the same scale as input.